spv/sync.go

```diff
@@ -1338,6 +1338,22 @@ func (s *Syncer) getHeaders(ctx context.Context, rp *p2p.RemotePeer) error {
 			}
 		}
 
+		// Verify the sidechain that includes the received headers has
+		// the correct difficulty.
+		s.sidechainMu.Lock()
+		fullsc, err := s.sidechains.FullSideChain(nodes)
+		if err != nil {
+			s.sidechainMu.Unlock()
+			return err
+		}
+		_, err = s.wallet.ValidateHeaderChainDifficulties(ctx, fullsc, 0)
```

Comment:
This seems fine for now, if only because ValidateHeaderChainDifficulties, despite using a slice of BlockNodes, only actually checks the headers and doesn't need the cfilters to be populated.  Since we have the chance to change the API, we may want to change that to take a slice of headers instead (can be in a future change).